### PR TITLE
chore(android): add CodeBuild project for release automation

### PR DESCRIPTION
**Pull request by rjuliano**
*Issue #, if available:*

*Description of changes:*
CodeBuild project for the buildspec file defined in [this PR](https://github.com/aws-amplify/amplify-android/pull/1271)

This PR adds a new stack with a CodeBuild project triggered by PR merges that meet certain base branch, source branch and commit message requirements.

It can also be triggered manually via the console or the AWS CLI.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/build_infrastructure/android/app.py [hunk 1]
```diff
@@ -41,5 +46,18 @@
...
 
+# Maven publisher stack
+maven_publisher_stack_props = {
+    'github_source': { 
+        'owner': github_owner,
+        'repo': REPO, 
+        'base_branch': branch,
+        'release_pr_branch': release_pr_branch
+    },
+    'codebuild_project_name_prefix': 'AmplifyAndroid'
```
Comment (richardmcclellan) [inline src/build_infrastructure/android/app.py:57]: Any way to make this agnostic of project, so that we can use this script for all 3 of our repos?

Comment (rjuliano) [reply src/build_infrastructure/android/app.py:57]: We'll need separate stacks, but we're reusing the CDK constructs.

When we setup the other projects, we'll add a `sdk_maven_publisher_stack_props` that will have a different CodeBuild project name prefix...something like `AndroidSDK`. Then we create the stack:

```python
MavenReleaseStack(app, "AndroidSDKMavenPublisher", sdk_maven_publisher_stack_props, description="Assets used for publishing aws-sdk-android to maven.", env=TARGET_ENV)
```

File: src/build_infrastructure/android/amplify_custom_resources/maven_release_publisher.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                    github_owner,
+                    github_repo,
+                    buildspec_path,
+                    environment_variables = {},
+                    base_branch: str = "main",
+                    release_branch: str = "release"):
+
+        build_environment = BuildEnvironment(build_image=self.BUILD_IMAGE, privileged = True, compute_type = ComputeType.LARGE)
+
+        trigger_on_pr_merged = FilterGroup.in_event_of(EventAction.PULL_REQUEST_MERGED).and_base_branch_is(base_branch).and_branch_is(release_branch).and_commit_message_is("chore(release).*")
```
Comment (richardmcclellan) [inline src/build_infrastructure/android/amplify_custom_resources/maven_release_publisher.py:27]:
```suggestion
        trigger_on_pr_merged = FilterGroup.in_event_of(EventAction.PULL_REQUEST_MERGED).and_base_branch_is(base_branch).and_branch_is(release_branch).and_commit_message_is("release:.*")
```

File: src/build_infrastructure/android/amplify_custom_resources/maven_release_publisher.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+class MavenPublisher(Project):
+    BUILD_IMAGE = LinuxBuildImage.AMAZON_LINUX_2_3
+    def __init__(self, scope: core.Construct, id: str, *,
+                    project_name: str,
+                    github_owner,
+                    github_repo,
+                    buildspec_path,
+                    environment_variables = {},
+                    base_branch: str = "main",
+                    release_branch: str = "release"):
```
Comment (richardmcclellan) [inline src/build_infrastructure/android/amplify_custom_resources/maven_release_publisher.py:23]:
```suggestion
                    release_branch: str = "bump_version"):
```

**Review by richardmcclellan — APPROVED** [review]

**Pull request merged**